manifests/service.pp

```diff
@@ -4,6 +4,21 @@
 class candlepin::service {
   assert_private()
 
+  if $candlepin::use_container {
+    $container_context = {
+      'image' => $candlepin::container_image,
+    }
+
+    file { '/etc/containers/systemd/tomcat.container':
```

Comment:
I'd also be tempted to try https://github.com/southalc/podman/pull/83. Comparing the two, your code doesn't take care of the `daemon-reload` that's needed when it changes. If we end up doing more with quadlets then it'll reduce the duplication.